File: leetcode/maximum-subarray/MaximumSubarray.java
public class MaximumSubarray {
    public static void main(String[] args) {
        
    }

    public class Solution {
        public int maxSubArray(int[] nums) {
            int MAX = Integer.MIN_VALUE;
            int sum = 0;
            for (int i = 0; i < nums.length; i++) {
                sum += nums[i];
                MAX = Math.max(MAX, sum);
                if (sum < 0) {
                    sum = 0;
                }
            }
            return MAX;
        }
    }
}

File: leetcode/maximum-subarray/maximum_subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        MAX = nums[0]
        sum = nums[0]
        for i in range(1, len(nums)):
            sum = max(nums[i], sum + nums[i])
            MAX = max(MAX, sum)
        return MAX